substrate/frame/support/test/tests/pallet_ui/config_metadata_non_type_info.stderr

```diff
@@ -0,0 +1,14 @@
+error[E0277]: the trait bound `<T as pallet::Config>::MyNonScaleTypeInfo: TypeInfo` is not satisfied
+  --> tests/pallet_ui/config_metadata_non_type_info.rs:29:8
+   |
+29 |         type MyNonScaleTypeInfo;
+   |              ^^^^^^^^^^^^^^^^^^ the trait `TypeInfo` is not implemented for `<T as pallet::Config>::MyNonScaleTypeInfo`
+   |
```

Comment:
I see what you mean now! Indeed, have added an extra check while collecting the metadata, we should have now a bit clearer error message here:

```bash
error: Invalid #[pallet::include_metadata] in #[pallet::config], collected type `MyNonScaleTypeInfo` does not implement `scale::TypeInfo`
  --> tests/pallet_ui/config_metadata_non_type_info.rs:28:4

```

Reply:
actually what I imagine was something like:

```rust
let span = /* span of `#[pallet::include_metadata]"  or otherwise call site for automatic inclusion*/
quote_spanned(span => /* expansion related to include metadata or automatic inclusion*/)
```

But everything is good to me, I don't have much opinion.
Your current implementation is more restrictive but more explicit so it can be the best.